src/ProjectsAccessProps.ts

```diff
@@ -41,18 +41,51 @@ export interface ProjectsSearchArg {
   exactMatch: boolean;
 }
 
-/** Set of optional arguments used for methods that allow advanced queries
+/** Possible Project sources.
  * @beta
  */
-export interface ProjectsQueryArg {
-  pagination?: ProjectsPaginationArg;
-  search?: ProjectsSearchArg;
+export enum ProjectsSource {
+  All = "",
+  Favorites = "favorites",
+  Recents = "recents"
 }
 
+/** Set of optional arguments used for methods that allow advanced queries.
+ * The source and search properties are mutually exclusive.
+ * @beta
+ */
+export type ProjectsQueryArg =
+  { pagination?: ProjectsPaginationArg, search?: never, source?: ProjectsSource } |
+  { pagination?: ProjectsPaginationArg, search?: ProjectsSearchArg, source?: never };
+
 /** Methods for accessing projects
  * @beta
 */
 export interface ProjectsAccess {
   /** Get projects associated with the requester */
   getAll(accessToken: AccessToken, arg?: ProjectsQueryArg): Promise<Project[]>;
+  /** Get projects and links by query */
+  getByQuery(accessToken: AccessToken, queryArg?: ProjectsQueryArg): Promise<ProjectsQueryResult>;
```

Comment:
Do we need to update the main interface? This is technically a breaking change to the API by enforcing a new method to be implemented.

Reply:
Also, can you align with the `getAll` signature and use `arg` instead of `queryArg`?

Reply:
OK, I removed the new method in the interface and changed the signature of the getByQuery function.